### ISSUE TITLE
New feature: auto-generate .manifest file

### ISSUE THREAD

**Issue by manavortex** [WolvenKit/WolvenKit#1434]
I figured we could discuss this here, rather than having it in three different Discord channels. I'm drafting a **proposal**, let's work out something cool, as not all prerequisites are currently implemented to satisfaction.

As mentioned by [Ineptus](https://discord.com/channels/717692382849663036/727800260486758410/1174797119157190786), it could be cool to have .manifest files in mods. Redmod already kinda provides something like that, but vanilla stuff doesn't.

# Motivation
Having this file could
- help users keeping track of what they have installed
- allow us to have a semi-automatic uninstaller ("drag a manifest on this .bat file")
- motivate us to finally fix up Wolvenkit project properties and come up with a proper GUI for that

# Included information
The .manifest files could contain the following information: 
- name of mod
- name of modder
- version of mod
- file structure under `packed`

# File stuff
## Target location
Suggestion: `packed\.mod_manifests` (so that it goes to the game root dir)
## File name
Suggestion: `<$WolvenkitProjectName>.manifest.txt`

# Prerequisites
What do we need to do before we can have this?
## Mandatory fields in WKit project
- mod author name
- version number
It would be cool to let the user configure Wolvenkit-wide defaults, at least for the author name (and their e-mail?).
Version numbers should default to `1.0.0` (be pre-filled?). 
=> Could/should we enforce semantic versioning?
## "Pack new version"
An easy way of bumping said version number, maybe via a new "Pack" variant?

**Comment by Ineptus** [comment]
Or could they go to a directory like `/aboutMods/<$WolvenkitProjectName>/` along with any other files (e.g. licence.txt, thanks.txt) an author would like to include? And what do you think about including dependencies/requirements as well?

**Comment by manavortex** [comment]
We'd have to add a field in the project settings for those - which would probably be a free text field, but... would be doable

**Comment by Simarilius-uk** [comment]
Version numbers should default to <1, so that half baked ones and first release are easily identifiable, you should need to manually set 1 for your first release, otherwise every iteration up to that point and 1.0 will be indistinguisable.

**Comment by manavortex** [comment]
Agree, I changed the "requirement" to 0.0.1. Maybe we can think up a nice workflow for users this way?

**Comment by effs** [comment]
If possible, I'd try to use the redmod convention and format.

Dependency specification will always be out of date unless it's used somewhere; until that it is more likely to be a net negative.
